query-service/schema/components/api-types.raml

```diff
@@ -0,0 +1,207 @@
+#%RAML 1.0 Library
+
+uses:
+  base: ./blast/blast-cli.raml
+
+types:
+  QueryJobRequest:
+    displayName: Query Job Request
+    type: object
+    additionalProperties: false
+    properties:
+      jobConfig:
+        displayName: Job Config
+        description: Job Configuration.
+        type: QueryJobConfig
+      blastConfig:
+        displayName: Blast Tool Config
+        description: |
+          Blast Tool Configuration.
+          
+          Implemented by one of:
+          
+          - [BlastNConfig](#type:lib.BlastNConfig)
+          - [BlastPConfig](#type:lib.BlastPConfig)
+          - [BlastXConfig](#type:lib.BlastXConfig)
+          - [DeltaBlastConfig](#type:lib.DeltaBlastConfig)
+          - [PSIBlastConfig](#type:lib.PSIBlastConfig)
+          - [RPSBlastConfig](#type:lib.RPSBlastConfig)
+          - [RPSTBlastNConfig](#type:lib.RPSTBlastNConfig)
+          - [TBlastNConfig](#type:lib.TBlastNConfig)
+          - [TBlastXConfig](#type:lib.TBlastXConfig)
+        type: base.BlastQueryConfig
+      meta:
+        displayName: User Metadata
+        type: QueryJobRequestMeta
+        required: false
+        default: null
+
+  QueryJobConfig:
+    type: object
+    additionalProperties: false
+    properties:
+      site:
+        displayName: Target Site
+        type: TargetSite
+      targets:
+        displayName: Query Targets
+        type: array
+        items:
+          type: QueryJobTarget
+        minItems: 1
+      query:
+        displayName: Blast Query
+        description: |
+          **For Requests**
+          
+          Inline query definition.
+          
+          This field is to be used for non-file-upload blast queries.
+          
+          If this field is not set, a query file must be uploaded for the
+          request to be valid.
+          
+          **For Responses**
+          
+          This field will be omitted.
+        type: string
+        required: false
+      addToCollection:
```

Comment:
I like "addToUserCollection" which I also like for above for explicitness.  Maybe "addToMyCollection" instead so we know -which- user, but I like the former more.

Reply:
Updated